### ISSUE TITLE
bug: "No such host is known" when there is no "/" at the end of the bucket name

### ISSUE THREAD

**Issue by mahgo** [localstack/localstack#7652]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When using localstack, with the AWS .net core SDK, calling any apis, I need to add a "/" on the end of my bucket name, otherwise I get the following error: 
```
System.Net.Http.HttpRequestException : No such host is known. (bucketname.localhost:4566)
```

When using an actual AWS S3 bucket, this is not required.

### Expected Behavior

APIs should succeed without needing to add "/" to end of bucket name.

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)
docker run --rm -it -p 4566:4566 -p 4510-4559:4510-4559 localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)
Setup:
```cs 
AmazonS3Config amazonS3Config = new AmazonS3Config
{
    ServiceURL = "http://localhost:4566"
};
AWSCredentials awsCredentials = new BasicAWSCredentials("foo", "bar");
AmazonS3Client amazonS3Client = new AmazonS3Client(awsCredentials, amazonS3Config);
```

This fails (throws exception):
```cs
await AmazonS3Util.DoesS3BucketExistV2Async(amazonS3Client, "bucketname")
```

This succeeds:
```cs
await AmazonS3Util.DoesS3BucketExistV2Async(amazonS3Client, "bucketname/")
```

### Environment

```markdown
- OS: Ubuntu 20.04, on WSL2 - Windows 11
- LocalStack: latest
```

**Comment by bentsku** [comment]
Hi @mahgo and thanks for your report.

As seen on the error message (`bucketname.localhost:4566`), the SDK uses the [virtual host addressing style](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html) by default. However, LocalStack does not support having the bucket prefixed before `localhost` directly when the endpoint configured is `localhost:4566`.

There are two solutions to this, related to the [configuration of your S3 Client](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/S3/TS3Config.html) in the SDK:

- Use a different endpoint for LocalStack, specific for s3, which accepts having the bucket in the host: `http://s3.localhost.localstack.cloud:4566` as your `ServiceURL`. I'm not an expert in .NET, so bear with me, but it could look like this.
```csharp
var config = new AmazonS3Config { ServiceURL = "http://s3.localhost.localstack.cloud:4566" };
var s3client = new AmazonS3Client(config);
```
This endpoint will accept the host addressed bucket request, the request URL will look like:
`http://{bucket-name}.s3.localhost.localstack.cloud`
We also have some [documentation about setting the endpoint with the AWS .NET SDK.](https://docs.localstack.cloud/user-guide/integrations/sdks/dotnet/)

- Second solution, use the path addressing style, making the bucket part of the path instead. 
```csharp
var config = new AmazonS3Config { ForcePathStyle = true };
var s3client = new AmazonS3Client(config);
```
This will force the SDK to use the path style, having your pre-signed URL look like:
`http://localhost:4566/{bucket-name}`

Hope that solves your issue, I'd appreciate your feedback. Thanks!

**Comment by bentsku** [comment]
Closing as it solved the issue as replied in #7651. Thanks again for your report!
